nnvm/src/top/tensor/reduce.cc

```diff
@@ -233,6 +255,40 @@ NNVM_REGISTER_REDUCE_OP(min)
     };
 });
 
+NNVM_REGISTER_COLLAPSE_OP(sum)
+.describe(R"code(Reduces lhs to the shape of rhs via sum)code" NNVM_ADD_FILELINE)
+.set_attr<FTVMCompute>(
+  "FTVMCompute", [](const NodeAttrs& attrs,
+                    const Array<Tensor>& inputs,
+                    const Array<Tensor>& out_info) {
+    auto ishape = topi::detail::GetConstIntValues(inputs[0]->shape, "ishape");
+    auto oshape = topi::detail::GetConstIntValues(inputs[1]->shape, "oshape");
+    CHECK_GE(ishape.size(), oshape.size()) << attrs.name;
+    std::vector<dim_t> r_axes;
+    bool keepdims = false;
+    std::vector<dim_t> squeeze_axes;
+    for (int i = ishape.size() - 1, j = oshape.size() - 1; i >= 0; --i) {
+      if (j < 0 || ishape[i] != oshape[j]) {
+        r_axes.push_back(i);
+        if (j < 0) {
+          squeeze_axes.push_back(i);
+        } else {
+          keepdims |= oshape[j] == 1;
+          j -= oshape[j] == 1;
+        }
+      } else {
+        --j;
+      }
```

Comment:
I prefer
```cpp
if ( j < 0 ) {
  r_axes.push_back(i);
  squeeze_axes.push_back(i);
} else if ( ishpae[i] != oshape[j]) {
  r_axes.push_back(i);
  if (oshape[j] == 1) {
    keepdims = true;
    --j;
  }
} else {
  --j;
}
```

Reply:
Just a bit. I changed the index variable names and added some comments to help make it clearer.

Reply:
looks good to me